domain/annotation/state/state_test.go

```diff
@@ -0,0 +1,404 @@
+// annotation_model table
+func (s *stateSuite) ensureAnnotation(c *gc.C, id, uuid, key, value string) {
+	if id == "model" {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			_, err := tx.ExecContext(ctx, `
+				INSERT INTO annotation_model (key, value)
+				VALUES (?, ?)`, key, value)
+			return err
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	} else {
+		err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+			_, err := tx.ExecContext(ctx, fmt.Sprintf(`
+				INSERT INTO annotation_%[1]s (uuid, key, value)
+				VALUES (?, ?, ?)`, id), uuid, key, value)
+			return err
+		})
+		c.Assert(err, jc.ErrorIsNil)
+	}
+}
+
+// ensureMachine manually inserts a row into the machine table.
+func (s *stateSuite) ensureMachine(c *gc.C, id, uuid string) {
+	s.ensureNetNode(c, "node2")
+	s.ensureLife(c, "life3", "3")
+
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+		INSERT INTO machine (uuid, net_node_uuid, machine_id, life_id)
+		VALUES (?, "node2", ?, "life3")`, uuid, id)
+		return err
+	})
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+// ensureApplication manually inserts a row into the application table.
+func (s *stateSuite) ensureApplication(c *gc.C, name, uuid string) {
+	s.ensureLife(c, "life3", "3")
+
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+		INSERT INTO application (uuid, name, life_id)
+		VALUES (?, ?, "life3")`, uuid, name)
+		return err
+	})
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+// ensureUnit manually inserts a row into the unit table.
+func (s *stateSuite) ensureUnit(c *gc.C, unit_id, uuid string) {
+	s.ensureApplication(c, "myapp", "123") // ensureApplication auto inserts life3
+	s.ensureNetNode(c, "321")
+
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+		INSERT INTO unit (uuid, unit_id, application_uuid, net_node_uuid, life_id)
+		VALUES (?, ?, ?, ?, ?)`, uuid, unit_id, "123", "321", "life3")
+		return err
+	})
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+// ensureCharm manually inserts a row into the charm table.
+func (s *stateSuite) ensureCharm(c *gc.C, url, uuid string) {
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+		INSERT INTO charm (uuid, url)
+		VALUES (?, ?)`, uuid, url)
+		return err
+	})
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+// ensureStorage inserts a row into the storage_instance table
+func (s *stateSuite) ensureStorage(c *gc.C, name, uuid string) {
+	s.ensureLife(c, "life3", "3")
+
+	// ensure storage_kind
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `
+		INSERT INTO storage_kind (id, kind)
```

Comment:
I would just use one of the kinds already in the table. In any case, there's no need to run separate transactions.

Reply:
Ah yes I didn't see the INSERTs for the `storage_kind` (neither for the `life` you mentioned below), I'll use those for sure 👍